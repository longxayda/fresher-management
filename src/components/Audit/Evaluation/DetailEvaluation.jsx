import React from "react";
import { useSelector } from "react-redux";
import { Detail, DetailLoading } from "redux/selectors/auditorsSelector/evaluationSelectors";
import { EvaluationTable } from "components/Audit/Evaluation/EvaluationTable";

const obj = [
    {
        Header: "Time",
        accessor: "time",
        Filter: () => null,
    },
    {
        Header: "Question",
        accessor: "content",
        Filter: () => null,
    },
    {
        Header: "Question Level",
        accessor: "level",
        Filter: () => null,
    },
    {
        Header: "Comment",
        accessor: "assessment",
        Filter: () => null,
    },
    {
        Header: "Score",
        accessor: "score",
        Filter: () => null,
    },
    {
        Header: "Auditor",
        accessor: "auditorName",
        Filter: () => null,
    },
];

const DetailEvaluation = () => {
    const columns = React.useMemo(() => obj, []);
    const data = useSelector(Detail);
    return (
        <>
            <EvaluationTable
                id="detail-component"
                columns={columns}
                data={data.listEva || []}
                currentIndex={1}
                maxIndex={1}
                setCurrentIndex={() => { }}
                load={DetailLoading}
            />
        </>

    );
}

export default DetailEvaluation;
