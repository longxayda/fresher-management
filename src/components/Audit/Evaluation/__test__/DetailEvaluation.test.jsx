import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, mount, render } from 'enzyme';

import { WrapTable } from '../DetailEvaluation';
import DetailEvaluation from '../DetailEvaluation';

Enzyme.configure({ adapter: new Adapter() });

describe('Test WrapTable', () => {
    test('renders WrapTable of DetailEvaluation with EMPTY props', () => { 
        const column = [], data = [];
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(false);
    });

    test('renders WrapTable of DetailEvaluation with NULL props', () => {
        const column = null, data = null;
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(false);
    });

    test('renders WrapTable of DetailEvaluation with INVALID VALUE 1 props', () => {
        const column = [1, 2], data = [3, 5];
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(false);
    });

    test('renders WrapTable of DetailEvaluation with INVALID VALUE 2 props', () => {
        const column = [{abv: 'asdas', as: 'afaa'}], data = [[{abv: 'asdas', as: 'afaa'}]];
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(false);
    });

    test('renders WrapTable of DetailEvaluation with INVALID TYPE props', () => {
        const column = 1, data = 1;
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(false);
    });

    test('renders WrapTable of DetailEvaluation with VALID VALUE props', () => {
        const column = [{Header: 'Name', accessor: 'name'}], data = [[{name: 'asdas'}]];
        const tree = shallow(<WrapTable columns={column} data={data} />);
        expect(tree.find('.wraptable-detail').exists()).toEqual(true);
    });
});

describe('Test DetailEvaluation', () => {
    test('check DetailEvaluation\'s component is EXIST', () => {
        const tree = mount(<DetailEvaluation />);
        expect(tree.find('.detail-evaluation').exists()).toEqual(true);
    });

    test('render DetailEvaluation\'s component correctly', () => {
        const tree = shallow(<DetailEvaluation />);
        expect(tree.getElements()).toMatchSnapshot();
    });
});