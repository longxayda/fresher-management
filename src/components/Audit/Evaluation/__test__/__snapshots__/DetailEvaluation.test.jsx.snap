// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DetailEvaluation render DetailEvaluation's component correctly 1`] = `
Array [
  <div
    className="detail-evaluation"
  >
    <WrapTable
      columns={
        Array [
          Object {
            "Filter": [Function],
            "Header": "Date",
            "accessor": "date",
          },
          Object {
            "Filter": [Function],
            "Header": "Question",
            "accessor": "question",
          },
          Object {
            "Filter": [Function],
            "Header": "Question Level",
            "accessor": "questionLevel",
          },
          Object {
            "Filter": [Function],
            "Header": "Comment",
            "accessor": "comment",
          },
          Object {
            "Filter": [Function],
            "Header": "Evaluate",
            "accessor": "evaluate",
          },
          Object {
            "Filter": [Function],
            "Header": "Auditor",
            "accessor": "auditor",
          },
        ]
      }
      data={
        Array [
          Object {
            "auditor": "Hortense",
            "comment": "\\"_-RxI|VPP",
            "date": "2022-07-24T14:39:42.681Z",
            "evaluate": 71687,
            "id": "1",
            "question": "Boston's most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
            "questionLevel": 93888,
          },
          Object {
            "auditor": "Emmalee",
            "comment": "xoyRE:l7%0",
            "date": "2022-07-25T09:05:59.431Z",
            "evaluate": 2859,
            "id": "2",
            "question": "Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals",
            "questionLevel": 61953,
          },
          Object {
            "auditor": "Monte",
            "comment": ",JqB6W&\\"(K",
            "date": "2022-07-25T01:42:30.620Z",
            "evaluate": 61591,
            "id": "3",
            "question": "The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J",
            "questionLevel": 97979,
          },
          Object {
            "auditor": "Claude",
            "comment": "UE\\\\l6R<P?J",
            "date": "2022-07-25T06:33:54.095Z",
            "evaluate": 61461,
            "id": "4",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 42549,
          },
          Object {
            "auditor": "Toby",
            "comment": "}18c^uPS,%",
            "date": "2022-07-25T06:19:51.227Z",
            "evaluate": 33996,
            "id": "5",
            "question": "Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals",
            "questionLevel": 16871,
          },
          Object {
            "auditor": "Jeffrey",
            "comment": "Oov0j:cHS1",
            "date": "2022-07-24T21:37:33.184Z",
            "evaluate": 63469,
            "id": "6",
            "question": "The beautiful range of Apple Naturalé that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients",
            "questionLevel": 79481,
          },
          Object {
            "auditor": "Esmeralda",
            "comment": "\\"wqdyBEPA0",
            "date": "2022-07-25T04:21:41.591Z",
            "evaluate": 75228,
            "id": "7",
            "question": "The Football Is Good For Training And Recreational Purposes",
            "questionLevel": 68277,
          },
          Object {
            "auditor": "Kasey",
            "comment": "Ov^fp}$|^=",
            "date": "2022-07-24T19:54:38.766Z",
            "evaluate": 80481,
            "id": "8",
            "question": "The Football Is Good For Training And Recreational Purposes",
            "questionLevel": 19758,
          },
          Object {
            "auditor": "Deanna",
            "comment": "gO-$[dmnY@",
            "date": "2022-07-24T11:35:56.162Z",
            "evaluate": 96976,
            "id": "9",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 89923,
          },
          Object {
            "auditor": "Gay",
            "comment": "ML,qLu=$eM",
            "date": "2022-07-24T19:50:02.206Z",
            "evaluate": 35368,
            "id": "10",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 99617,
          },
          Object {
            "auditor": "Will",
            "comment": "s}:,%sOLQ*",
            "date": "2022-07-24T09:28:31.266Z",
            "evaluate": 89567,
            "id": "11",
            "question": "The beautiful range of Apple Naturalé that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients",
            "questionLevel": 16772,
          },
          Object {
            "auditor": "Mauricio",
            "comment": "wOz!F[;v\\\\|",
            "date": "2022-07-24T23:24:29.878Z",
            "evaluate": 90450,
            "id": "12",
            "question": "The beautiful range of Apple Naturalé that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients",
            "questionLevel": 47319,
          },
          Object {
            "auditor": "Cortney",
            "comment": "2W$J1p>u{7",
            "date": "2022-07-25T02:47:19.332Z",
            "evaluate": 65848,
            "id": "13",
            "question": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
            "questionLevel": 27734,
          },
          Object {
            "auditor": "Antonio",
            "comment": ";a&u.!B?qG",
            "date": "2022-07-24T15:43:28.329Z",
            "evaluate": 85948,
            "id": "14",
            "question": "Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals",
            "questionLevel": 70583,
          },
          Object {
            "auditor": "Edd",
            "comment": "f6EV\\"Lxt4/",
            "date": "2022-07-25T01:00:23.872Z",
            "evaluate": 11752,
            "id": "15",
            "question": "The Football Is Good For Training And Recreational Purposes",
            "questionLevel": 75069,
          },
          Object {
            "auditor": "Connor",
            "comment": "M^F6+0c\`MJ",
            "date": "2022-07-25T03:33:04.804Z",
            "evaluate": 777,
            "id": "16",
            "question": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
            "questionLevel": 53533,
          },
          Object {
            "auditor": "Lea",
            "comment": "t-=lGSmqNi",
            "date": "2022-07-25T06:19:44.004Z",
            "evaluate": 95983,
            "id": "17",
            "question": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
            "questionLevel": 36245,
          },
          Object {
            "auditor": "Florence",
            "comment": "=y\\\\czr#uqY",
            "date": "2022-07-24T23:00:52.760Z",
            "evaluate": 74708,
            "id": "18",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 85770,
          },
          Object {
            "auditor": "Dorthy",
            "comment": "Xaq>Z_Qo>B",
            "date": "2022-07-24T09:43:09.984Z",
            "evaluate": 89110,
            "id": "19",
            "question": "Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals",
            "questionLevel": 27089,
          },
          Object {
            "auditor": "Otilia",
            "comment": "G(o#15>@Rj",
            "date": "2022-07-24T17:53:13.862Z",
            "evaluate": 24384,
            "id": "20",
            "question": "The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J",
            "questionLevel": 28536,
          },
          Object {
            "auditor": "Christopher",
            "comment": "<<ZIn%x<BU",
            "date": "2022-07-25T00:33:45.619Z",
            "evaluate": 7647,
            "id": "21",
            "question": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
            "questionLevel": 92365,
          },
          Object {
            "auditor": "Walton",
            "comment": "<P@3SZYxoU",
            "date": "2022-07-25T01:28:36.983Z",
            "evaluate": 52871,
            "id": "22",
            "question": "The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive",
            "questionLevel": 81179,
          },
          Object {
            "auditor": "Nannie",
            "comment": "\`!WLly<*SM",
            "date": "2022-07-25T06:41:35.182Z",
            "evaluate": 97524,
            "id": "23",
            "question": "New ABC 13 9370, 13.3, 5th Gen CoreA5-8250U, 8GB RAM, 256GB SSD, power UHD Graphics, OS 10 Home, OS Office A & J 2016",
            "questionLevel": 46701,
          },
          Object {
            "auditor": "Jerad",
            "comment": "S,qXe>)6S%",
            "date": "2022-07-25T06:56:41.115Z",
            "evaluate": 85093,
            "id": "24",
            "question": "Carbonite web goalkeeper gloves are ergonomically designed to give easy fit",
            "questionLevel": 99806,
          },
          Object {
            "auditor": "Layla",
            "comment": "Y_7@+%OE*V",
            "date": "2022-07-25T08:09:23.699Z",
            "evaluate": 94744,
            "id": "25",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 103,
          },
          Object {
            "auditor": "Doyle",
            "comment": "\\"POKe(Pt?C",
            "date": "2022-07-25T00:04:03.569Z",
            "evaluate": 39845,
            "id": "26",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 97433,
          },
          Object {
            "auditor": "Sandy",
            "comment": "'$fVBYY$Y9",
            "date": "2022-07-24T15:42:36.615Z",
            "evaluate": 8484,
            "id": "27",
            "question": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
            "questionLevel": 43738,
          },
          Object {
            "auditor": "Angeline",
            "comment": "(X1>3C<d>=",
            "date": "2022-07-24T12:28:51.238Z",
            "evaluate": 69444,
            "id": "28",
            "question": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
            "questionLevel": 50116,
          },
          Object {
            "auditor": "Katelynn",
            "comment": "YL0o5)s\\\\p=",
            "date": "2022-07-24T18:35:11.719Z",
            "evaluate": 70351,
            "id": "29",
            "question": "Carbonite web goalkeeper gloves are ergonomically designed to give easy fit",
            "questionLevel": 12764,
          },
          Object {
            "auditor": "Hassan",
            "comment": "_C<Zf<*&'q",
            "date": "2022-07-25T03:24:30.849Z",
            "evaluate": 44077,
            "id": "30",
            "question": "Boston's most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
            "questionLevel": 94657,
          },
          Object {
            "auditor": "Aurelio",
            "comment": ".q+RIw3jI?",
            "date": "2022-07-25T01:48:34.459Z",
            "evaluate": 41552,
            "id": "31",
            "question": "The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive",
            "questionLevel": 65373,
          },
          Object {
            "auditor": "Merlin",
            "comment": "&hNb_IqLbq",
            "date": "2022-07-24T15:58:06.513Z",
            "evaluate": 27504,
            "id": "32",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 25232,
          },
          Object {
            "auditor": "Karli",
            "comment": "<L}gRlOWT;",
            "date": "2022-07-25T00:01:40.854Z",
            "evaluate": 3714,
            "id": "33",
            "question": "The Football Is Good For Training And Recreational Purposes",
            "questionLevel": 54867,
          },
          Object {
            "auditor": "Zula",
            "comment": "_fi[qp@dF8",
            "date": "2022-07-24T19:13:46.886Z",
            "evaluate": 84604,
            "id": "34",
            "question": "The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive",
            "questionLevel": 51776,
          },
          Object {
            "auditor": "Luella",
            "comment": "1Z9g@sN/_T",
            "date": "2022-07-25T00:17:48.654Z",
            "evaluate": 54823,
            "id": "35",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 2664,
          },
          Object {
            "auditor": "Norene",
            "comment": "I1HDavhQtB",
            "date": "2022-07-24T23:18:40.331Z",
            "evaluate": 33516,
            "id": "36",
            "question": "New ABC 13 9370, 13.3, 5th Gen CoreA5-8250U, 8GB RAM, 256GB SSD, power UHD Graphics, OS 10 Home, OS Office A & J 2016",
            "questionLevel": 43584,
          },
          Object {
            "auditor": "Karlie",
            "comment": "}EKvSt(KrP",
            "date": "2022-07-24T13:17:45.163Z",
            "evaluate": 92965,
            "id": "37",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 25616,
          },
          Object {
            "auditor": "Cornelius",
            "comment": "WCLnG#YqUB",
            "date": "2022-07-25T08:08:44.578Z",
            "evaluate": 84679,
            "id": "38",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 75152,
          },
          Object {
            "auditor": "Greta",
            "comment": "/z/HPl,c\\"P",
            "date": "2022-07-24T21:23:36.851Z",
            "evaluate": 1959,
            "id": "39",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 26454,
          },
          Object {
            "auditor": "Kayden",
            "comment": "+8T'cj1V.E",
            "date": "2022-07-24T13:24:36.119Z",
            "evaluate": 12359,
            "id": "40",
            "question": "The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive",
            "questionLevel": 31575,
          },
          Object {
            "auditor": "Chelsey",
            "comment": "d<tw'tI^Uq",
            "date": "2022-07-25T02:03:19.996Z",
            "evaluate": 24950,
            "id": "41",
            "question": "The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J",
            "questionLevel": 83254,
          },
          Object {
            "auditor": "Vincent",
            "comment": "!i=AqEa80M",
            "date": "2022-07-24T18:08:47.202Z",
            "evaluate": 41859,
            "id": "42",
            "question": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
            "questionLevel": 97904,
          },
          Object {
            "auditor": "Rosamond",
            "comment": "=0#W^&]r_X",
            "date": "2022-07-24T11:05:53.369Z",
            "evaluate": 53417,
            "id": "43",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 24057,
          },
          Object {
            "auditor": "Jena",
            "comment": "B4;xW(4M1+",
            "date": "2022-07-24T10:55:45.727Z",
            "evaluate": 72568,
            "id": "44",
            "question": "The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J",
            "questionLevel": 75244,
          },
          Object {
            "auditor": "Olga",
            "comment": "4!JlX02$@W",
            "date": "2022-07-24T12:28:45.597Z",
            "evaluate": 63879,
            "id": "45",
            "question": "New ABC 13 9370, 13.3, 5th Gen CoreA5-8250U, 8GB RAM, 256GB SSD, power UHD Graphics, OS 10 Home, OS Office A & J 2016",
            "questionLevel": 39719,
          },
          Object {
            "auditor": "Christian",
            "comment": "P]0.7m{M\`f",
            "date": "2022-07-25T06:47:08.183Z",
            "evaluate": 11524,
            "id": "46",
            "question": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
            "questionLevel": 6500,
          },
          Object {
            "auditor": "Yasmin",
            "comment": "6NQGX_'X^Z",
            "date": "2022-07-25T03:15:09.011Z",
            "evaluate": 69063,
            "id": "47",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 58415,
          },
          Object {
            "auditor": "Sonya",
            "comment": "9+i6U(&TpS",
            "date": "2022-07-24T19:09:28.180Z",
            "evaluate": 8238,
            "id": "48",
            "question": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
            "questionLevel": 80392,
          },
          Object {
            "auditor": "Mckayla",
            "comment": "w]^TqF<@dE",
            "date": "2022-07-24T19:28:52.835Z",
            "evaluate": 11154,
            "id": "49",
            "question": "The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J",
            "questionLevel": 62441,
          },
          Object {
            "auditor": "Alvah",
            "comment": "xB$!q*DJ+M",
            "date": "2022-07-24T16:52:50.491Z",
            "evaluate": 59640,
            "id": "50",
            "question": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "questionLevel": 15363,
          },
        ]
      }
    />
  </div>,
]
`;
